cmake/DefaultTargetProperties.cmake

```diff
@@ -0,0 +1,206 @@
+
+    # TODO(DF): Currently set up to assume the build artifacts will
+    #   never be used directly, and the install artifacts have
+    #   everything they need in the standard search path(s). Hence we
+    #   expect CI/tests to use an install with all dependencies
+    #   available on standard search paths. This may turn out not to be
+    #   practical.
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        # Control RPATH in build phase.
+        SKIP_BUILD_RPATH TRUE
+        BUILD_WITH_INSTALL_RPATH FALSE
+        # Whether to add hardcoded directories to final runtime search
+        # path.
+        INSTALL_RPATH_USE_LINK_PATH FALSE
+        # Runtime search path value
+        INSTALL_RPATH "${rpath}"
+        # Enable RPATH on OSX
+        # TODO(DF): See above re. OSX RPATH
+        # MACOSX_RPATH ON
+    )
+
+    # CentOS 7 and below (i.e. VFX reference platform) compile gcc to
+    # only set RPATH by default, whereas other Linux distros set RUNPATH
+    # (if both are set then RPATH is ignored). They each cause different
+    # runtime search path behavior. So be explicit to ensure we are
+    # consistent across distros.
+    if (IS_GCC_OR_CLANG)
+        if (OPENASSETIO_ENABLE_NEW_DTAGS)
+            target_link_options(${target_name} PRIVATE "-Wl,--enable-new-dtags")
+        else ()
+            target_link_options(${target_name} PRIVATE "-Wl,--disable-new-dtags")
+        endif ()
+    endif ()
+
+
+    #-------------------------------------------------------------------
+    # C++ #define to toggle static vs dynamic lib
+
+    # Used by header generated by CMake's `generate_export_header`
+    if (NOT BUILD_SHARED_LIBS)
+        target_compile_definitions(${target_name} PRIVATE ${PROJECT_NAME}_STATIC_DEFINE=1)
+    endif ()
+
+
+    #-------------------------------------------------------------------
+    # Build tree directory structure.
+
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        # Make the build area layout look a bit more like the final dist
+        # layout
+        LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}
+        RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}
+    )
+
+
+    #-------------------------------------------------------------------
+    # Library version
+
+    # When building or installing appropriate symlinks are created, if
+    # supported.
+    set_target_properties(
+        ${target_name}
+        PROPERTIES
+        VERSION ${PROJECT_VERSION}
+        SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
+    )
+
+
+    #-------------------------------------------------------------------
+    # Coverage reports and/or sanitizers, useful for tests.
+
+    if (IS_GCC_OR_CLANG)
+        if (OPENASSETIO_ENABLE_COVERAGE)
+            target_compile_options(${target_name} PRIVATE --coverage -O0 -g)
```

Comment:
Is this worthy of a CMake message?

I'm thinking, what if a release build is made, coverage is enabled, and that gets shipped? Wouldn't be fun with debug symbols and no optimisations enabled.

Granted, very niche. But, if the option is there, people will use it.

Reply:
Added additional `STATUS` messages in 9ef4f8bf and added a `NOTICE` message specifically for coverage in fe213317